src/ansys/fluent/core/launcher/standalone_launcher.py

```diff
@@ -172,6 +184,12 @@ def __init__(
         The allocated machines and core counts are queried from the scheduler environment and
         passed to Fluent.
         """
+        if version == "2d" and gpu:
+            raise GPUSolverSupportError()
```

Comment:
@hpohekar I'll just take the above two lines as an example but this will apply generally to other code. We want to do this check not just for this launcher but for other launchers, too. And I see that you are already doing that, so that's good. But we shouldn't duplicate the code to do that in each launcher. We need to factor things out effectively.

Reply:
@seanpearsonuk I thought our requirement is to keep all launchers independent and therefore minimum code should be inside a common function like `create_launcher` therefore we have kept `_process_invalid_args` only there. So now to remove all duplication we have moved it to `create_launcher`

Reply:
If some subset (>1) of launchers all agree on a generalised validation for a given combination of argument values then each of those launchers should call a common function, e.g., 
```
def validate_gpu(gpu, version):
  if version == "2d" and gpu:
          raise GPUSolverSupportError()
```
Once we have written each launcher code in terms of a sequence of such high-level validation calls, we can look at what we have and see how it should be refactored further - there are options.

Reply:
So we want to add following function to pyfluent_enums.py and then we will call it in `create_launcher` right ?

```
def _validate_gpu(gpu, version):
  if version == "2d" and gpu:
          raise GPUSolverSupportError()
```

Reply:
My thinking is that it shall be possible to use each launcher independently which means that each one is in control of its argument validation.

Reply:
Ok. We will call it inside of each launcher. Thank you.

Reply:
Done. Thank you.